packages/react/src/components/PaginationNav/PaginationNav.tsx

```diff
@@ -0,0 +1,635 @@
+/**
+ * Copyright IBM Corp. 2020
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import PropTypes from 'prop-types';
+import React, { useState, useEffect, useRef } from 'react';
+import classnames from 'classnames';
+import {
+  CaretRight,
+  CaretLeft,
+  OverflowMenuHorizontal,
+} from '@carbon/icons-react';
+import { IconButton } from '../IconButton';
+import { usePrefix } from '../../internal/usePrefix';
+
+const translationIds = {
+  'carbon.pagination-nav.next': 'Next',
+  'carbon.pagination-nav.previous': 'Previous',
+  'carbon.pagination-nav.item': 'Page',
+  'carbon.pagination-nav.active': 'Active',
+  'carbon.pagination-nav.of': 'of',
+};
+
+function translateWithId(messageId: string): string {
+  return translationIds[messageId];
+}
+
+// https://reactjs.org/docs/hooks-faq.html#how-to-get-the-previous-props-or-state
+function usePrevious(value: number) {
+  const ref = useRef<number | null>(null);
+
+  useEffect(() => {
+    ref.current = value;
+  });
+
+  return ref.current;
+}
+
+function getCuts(
```

Comment:
@guidari I'm trying to think if there is a better name for this function. Thoughts??

Reply:
Maybe `calculateCuts()` would be nice, because it seems that is doing a few calculations there. 
I used as insparation the Menu.tsx that has the `calculatePosition()`

Reply:
yesss love it!

Reply:
@mattborghi can we update the function name to `calculateCuts` - thank you!

Reply:
Done!